drivers/sensor/fdc2x1x/fdc2x1x.c

```diff
@@ -0,0 +1,1010 @@
+	if (ret) {
+		return ret;
+	}
+
+#ifdef CONFIG_FDC2X1X_TRIGGER_NONE
+	/* Enable Data Ready Flag to poll for new measurement */
+	ret = fdc2x1x_reg_write_mask(dev, FDC2X1X_ERROR_CONFIG,
+				     FDC2X1X_ERROR_CONFIG_DRDY_2INT_MSK,
+				     FDC2X1X_ERROR_CONFIG_DRDY_2INT_SET(1));
+	if (ret) {
+		return ret;
+	}
+
+	/* INTB asserts by default, so disable it here */
+	ret = fdc2x1x_set_interrupt_pin(dev, false);
+	if (ret) {
+		return ret;
+	}
+#endif
+
+	ret = fdc2x1x_set_current_drv(dev, cfg->current_drv);
+	if (ret) {
+		return ret;
+	}
+
+	return 0;
+}
+
+/**
+ * Probe device (Check if it is the correct device).
+ * @param dev - The device structure.
+ * @return 0 in case of success, negative error code otherwise.
+ */
+static int fdc2x1x_probe(const struct device *dev)
+{
+	struct fdc2x1x_data *data = dev->data;
+	uint16_t dev_id, man_id;
+
+	if (fdc2x1x_reg_read(dev, FDC2X1X_DEVICE_ID, &dev_id) < 0) {
+		LOG_ERR("Failed to read device id");
+		return -EIO;
+	}
+
+	if (dev_id == FDC2X1X_DEVICE_ID_VAL_28BIT) {
+		data->fdc221x = true;
+	} else if (dev_id == FDC2X1X_DEVICE_ID_VAL) {
+		data->fdc221x = false;
+	} else {
+		LOG_ERR("Wrong device id");
+		return -ENODEV;
+	}
+
+	if (fdc2x1x_reg_read(dev, FDC2X1X_MANUFACTURER_ID, &man_id) < 0) {
+		LOG_ERR("Failed to read manufacturer id");
+		return -EIO;
+	}
+
+	if (man_id != FDC2X1X_MANUFACTURER_ID_VAL) {
+		LOG_ERR("Wrong manufaturer id");
+		return -ENODEV;
+	}
+
+	return 0;
+}
+
+/**
+ * Initialize the SD-Pin.
+ * @param dev - The device structure.
+ * @return 0 in case of success, negative error code otherwise.
+ */
+static int fdc2x1x_init_sd_pin(const struct device *dev)
+{
+	struct fdc2x1x_data *data = dev->data;
+	const struct fdc2x1x_config *cfg = dev->config;
+
+	data->gpio = device_get_binding(cfg->sd_name);
+	if (!data->gpio) {
+		LOG_ERR("Failed to get pointer to %s device!", cfg->sd_name);
+		return -EINVAL;
+	}
+
+	gpio_pin_configure(data->gpio, cfg->sd_pin,
+			   GPIO_OUTPUT_INACTIVE | cfg->sd_flags);
+
+	gpio_pin_set(data->gpio, cfg->sd_pin, 0);
+
+	return 0;
+}
+
+/**
+ * Initialization of the device.
+ * @param dev - The device structure.
+ * @return 0 in case of success, negative error code otherwise.
+ */
+static int fdc2x1x_init(const struct device *dev)
+{
+	struct fdc2x1x_data *data = dev->data;
+	const struct fdc2x1x_config *cfg = dev->config;
+	uint8_t ch_supported;
+
+	if (cfg->fdc2x14) {
+		ch_supported = 4;
+	} else {
+		ch_supported = 2;
+	}
+
+	if ((cfg->num_channels <= 0)) {
+		LOG_ERR("No channel nodes found");
+		return -EINVAL;
+	} else if (cfg->num_channels > ch_supported) {
+		LOG_ERR("Amount of channels not supported by this device");
+		return -EINVAL;
+	}
+
+	if (cfg->sd_name) {
+		if (fdc2x1x_init_sd_pin(dev) < 0) {
+			return -EINVAL;
+		}
+	}
+
+	data->bus = device_get_binding(cfg->i2c_name);
+	if (data->bus  == NULL) {
+		LOG_ERR("Failed to get pointer to %s device!", cfg->i2c_name);
+		return -EINVAL;
+	}
+
+	k_sleep(K_MSEC(100));
```

Comment:
Why? This is going to block the init sequence for 100 msec.

Reply:
Forgot to remove it after testing. Removed it now